src/bio/bio_wal.h

```diff
@@ -0,0 +1,100 @@
+/**
+ * (C) Copyright 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#ifndef __BIO_WAL_H__
+#define __BIO_WAL_H__
+
+#include "bio_internal.h"
+
+/* Meta blob header */
+struct meta_header {
+	uint32_t	mh_magic;
+	uint32_t	mh_version;
+	uuid_t		mh_meta_devid;		/* Meta SSD device ID */
+	uuid_t		mh_wal_devid;		/* WAL SSD device ID */
+	uuid_t		mh_data_devid;		/* Data SSD device ID */
+	uint64_t	mh_meta_blobid;		/* Meta blob ID */
+	uint64_t	mh_wal_blobid;		/* WAL blob ID */
+	uint64_t	mh_data_blobid;		/* Data blob ID */
+	uint32_t	mh_blk_bytes;		/* Block size for meta, in bytes */
+	uint32_t	mh_hdr_blks;		/* Meta blob header size, in blocks */
+	uint64_t	mh_tot_blks;		/* Meta blob capacity, in blocks */
+	uint32_t	mh_vos_id;		/* Associated per-engine target ID */
+	uint32_t	mh_padding1;		/* Reserved */
+	uint64_t	mh_padding2;		/* Reserved */
+	uint64_t	mh_padding3;		/* Reserved */
+	uint32_t	mh_padding4;		/* Reserved */
+	uint16_t	mh_csum_type;
+	uint16_t	mh_csum_len;
+	uint8_t		mh_csum[0];		/* Checksum of this header */
+};
+
+/* WAL blob header */
+struct wal_header {
+	uint32_t	wh_magic;
```

Comment:
could we add a wh_version here too, in case wal_header need be extended in the future.